main.py

```diff
@@ -0,0 +1,70 @@
+from fastapi import FastAPI, Depends, HTTPException
+from sqlalchemy.orm import Session
+
+import crud, database, schemas
+
+app = FastAPI()
+
+
+def get_db() -> Session:
+    db_session = database.SessionLocal()
+    try:
+        yield db_session
+    finally:
+        db_session.close()
+
+
+@app.get("/")
+def root() -> dict:
+    return {"Message": "Welcome to library"}
+
+
+@app.get("/authors/", response_model=list[schemas.AuthorInfo])
+def read_authors(
+    skip: int = 0, limit: int = 10, db_session: Session = Depends(get_db)
+):
+    return crud.get_all_authors(db_session=db_session)[skip : skip + limit]
+
+
+@app.post("/authors/", response_model=schemas.AuthorInfo)
+def create_author(
+    author: schemas.AuthorCreate, db_session: Session = Depends(get_db)
+):
+    db_existing_author = crud.get_author_by_name(
+        db_session=db_session, name=author.name
+    )
+    if db_existing_author:
+        raise HTTPException(
+            status_code=400, detail="Author with such name already exist"
```

Comment:
Consider rephrasing the error message to 'Author with this name already exists' for better clarity.